From fc630bcc025505c2d68a55a2de5829b852918235 Mon Sep 17 00:00:00 2001
From: aslammdkhan <aslam.khan@hpe.com>
Date: Fri, 3 Jun 2016 13:35:53 -0700
Subject: [PATCH 10/11] schema-to-idl populates value if it is present

The auto generation script is not looking for value part of key-value
and is not updating the value in set_subinterface idl api when min=0 and
max=1.  Added code to check if value is present fill that also.

Signed-off-by: aslammdkhan <aslam.khan@hpe.com>

---
 ovsdb/ovsdb-idlc.in | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/ovsdb/ovsdb-idlc.in b/ovsdb/ovsdb-idlc.in
index 36c5507..287df22 100755
--- a/ovsdb/ovsdb-idlc.in
+++ b/ovsdb/ovsdb-idlc.in
@@ -925,6 +925,8 @@ const struct ovsdb_datum *
                 txn_write_func = "ovsdb_idl_txn_write_clone"
             elif type.n_max == 1:
                 print "    union ovsdb_atom key;"
+                if type.value:
+                    print "    union ovsdb_atom value;"
                 print
                 print "    ovs_assert(inited);"
                 print "    if (%s) {" % nVar
@@ -935,7 +937,11 @@ const struct ovsdb_datum *
                 print "        datum.n = 0;"
                 print "        datum.keys = NULL;"
                 print "    }"
-                print "    datum.values = NULL;"
+                if type.value:
+                    print "    datum.values = &value;"
+                    print "    " + type.value.copyCValue("datum.values[0].%s" % type.value.type.to_string(), "%s[0]" % valueVar)
+                else:
+                    print "    datum.values = NULL;"
                 txn_write_func = "ovsdb_idl_txn_write_clone"
             else:
                 print "    size_t i;"
-- 
2.1.4

